lib/curl_fnmatch.c

```diff
@@ -245,7 +246,7 @@ static int setcharset(unsigned char **p, unsigned char *charset)
       if(c == ']')
         return SETCHARSET_OK;
       state  = CURLFNM_SCHS_DEFAULT;
-      charset[c] = 1;
+      charset[(unsigned int)c] = 1;
```

Comment:
Here too, you add **seven** typecasts. Why is this better?

Reply:
In case c is negative